### PR TITLE
Reflect Ingress and LB Class to remote clusters

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr allows to reflect the Ingresses and LoadBalance resources by changing the class in different clusters

# How Has This Been Tested?

- [x] locally

### CODE REVIEW

File: deployments/liqo/templates/liqo-controller-manager-deployment.yaml [hunk 1]
```diff
@@ -121,6 +121,10 @@ spec:
           - --real-storage-class-name={{ .Values.storage.realStorageClassName }}
           - --storage-namespace={{ .Values.storage.storageNamespace }}
           {{- end }}
+          {{- $d := dict "commandName" "--ingress-classes" "list" .Values.reflection.ingress.announcedIngressClasses }}
+          {{- include "liqo.concatenateListDefault" $d | nindent 10 }}
+          {{- $d := dict "commandName" "--load-balancer-classes" "list" .Values.reflection.service.announcedLoadBalancerClasses }}
+          {{- include "liqo.concatenateListDefault" $d | nindent 10 }}
```
Comment (cheina97) [inline deployments/liqo/templates/liqo-controller-manager-deployment.yaml:127]: What about using a separate flag to specify the default?

File: cmd/liqo-controller-manager/main.go [hunk 2]
```diff
@@ -157,6 +159,8 @@ func main() {
 	offerUpdateThreshold := argsutils.Percentage{}
 	flag.Var(&offerUpdateThreshold, "offer-update-threshold-percentage",
 		"The threshold (in percentage) of resources quantity variation which triggers a ResourceOffer update")
+	flag.Var(&ingressClasses, "ingress-classes", "List of ingress classes offered by the cluster. Example: \"nginx;default,traefik\"")
+	flag.Var(&loadBalancerClasses, "load-balancer-classes", "List of load balancer classes offered by the cluster. Example: \"nginx;default,traefik\"")
```
Comment (fra98) [inline cmd/liqo-controller-manager/main.go:163]:
```suggestion
	flag.Var(&loadBalancerClasses, "load-balancer-classes", "List of load balancer classes offered by the cluster. Example: \"metallb;default\"")
```

File: pkg/virtualKubelet/forge/ingresses.go [hunk 3]
```diff
@@ -22,11 +22,12 @@ import (
 )
 
 // RemoteIngress forges the apply patch for the reflected ingress, given the local one.
-func RemoteIngress(local *netv1.Ingress, targetNamespace string, forgingOpts *ForgingOpts) *netv1apply.IngressApplyConfiguration {
+func RemoteIngress(local *netv1.Ingress, targetNamespace string, enableIngress bool, remoteRealStorageClassName string,
```
Comment (fra98) [inline pkg/virtualKubelet/forge/ingresses.go:25]:
```suggestion
func RemoteIngress(local *netv1.Ingress, targetNamespace string, enableIngress bool, remoteRealIngressClassName string,
```

File: pkg/virtualKubelet/forge/ingresses.go [hunk 4]
```diff
@@ -36,11 +37,15 @@ func FilterIngressAnnotations(local map[string]string) map[string]string {
 
 // RemoteIngressSpec forges the apply patch for the specs of the reflected ingress, given the local one.
 // It expects the local object to be a deepcopy, as it is mutated.
-func RemoteIngressSpec(local *netv1.IngressSpec) *netv1apply.IngressSpecApplyConfiguration {
-	return netv1apply.IngressSpec().
+func RemoteIngressSpec(local *netv1.IngressSpec, enableIngress bool, remoteRealStorageClassName string) *netv1apply.IngressSpecApplyConfiguration {
```
Comment (fra98) [inline pkg/virtualKubelet/forge/ingresses.go:40]:
```suggestion
func RemoteIngressSpec(local *netv1.IngressSpec, enableIngress bool, remoteRealIngressClassName string) *netv1apply.IngressSpecApplyConfiguration {
```

File: pkg/virtualKubelet/forge/ingresses.go [hunk 5]
```diff
@@ -36,11 +37,15 @@ func FilterIngressAnnotations(local map[string]string) map[string]string {
...
+	ret := netv1apply.IngressSpec().
 		WithDefaultBackend(RemoteIngressBackend(local.DefaultBackend)).
 		WithRules(RemoteIngressRules(local.Rules)...).
 		WithTLS(RemoteIngressTLS(local.TLS)...)
+	if enableIngress {
+		ret.WithIngressClassName(remoteRealStorageClassName)
+	}
+	return ret
```
Comment (fra98) [inline pkg/virtualKubelet/forge/ingresses.go:48]:
```suggestion
	if enableIngress {
		ret.WithIngressClassName(remoteRealIngressClassName)
	} else if local.IngressClassName != nil {
		ret.WithIngressClassName(*local.IngressClassName)
	}
	return ret
```

**Review by fra98 — APPROVED** [review]

**Review by cheina97 — APPROVED** [review]

**Pull request merged**